### PR TITLE
new: update SDK to plugin API version 3.4.0

### PR DESCRIPTION
**Pull request by mrgian**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area plugin-sdk

> /area tests

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: include/falcosecurity/internal/symbols_extraction.h [hunk 1]
```diff
@@ -29,9 +29,11 @@ limitations under the License.
     static plugin_mixin<__t> s_plugin_extraction;                              \
                                                                                \
     FALCOSECURITY_EXPORT                                                       \
-    uint16_t* plugin_get_extract_event_types(uint32_t* numtypes)               \
+    uint16_t* plugin_get_extract_event_types(uint32_t* numtypes,               \
```
Comment (jasondellaluce) [inline include/falcosecurity/internal/symbols_extraction.h:32]: Shouldn't this apply to `symbols_parsing.h` as well?

File: examples/plugin_extract/plugin.cpp [hunk 2]
```diff
@@ -46,9 +46,18 @@ class my_plugin
     }
 
     // (optional)
-    void destroy() {}
+    bool set_config(falcosecurity::set_config_input& i) { return false; }
```
Comment (jasondellaluce) [inline examples/plugin_extract/plugin.cpp:49]: I suggest returning true and logging at least in one example plugin, just to check that everything works.

**Review by jasondellaluce — COMMENTED** [review]
Looks amazing!

File: include/falcosecurity/internal/symbols_extraction.h (hunk 1 above)
Comment (mrgian) [reply include/falcosecurity/internal/symbols_extraction.h:32]: Yep! Forgot about that

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**